src/components/heatmap/heatmap.tsx

```diff
@@ -84,32 +91,48 @@ const Heatmap = ({
         'hide_null_choice': true
       }]
     });
-
-    ConfigService.ERMrest.resolve(ermrestURI.ermrestUri, header).then((response: any) => {
-      const reference = response.contextualize.compact;
-      verifyConfiguration(reference);
-      if (!configErrorsPresent) {
-        const sortBy = typeof Object(config).data.sortBy !== 'undefined' ? Object(config).data.sortBy : [];
-        const ref = reference.sort(sortBy);
-        setHeader((prevHeader: any) => ({ ...prevHeader, action: 'main' }));
-        const pcid = getQueryParams(location.href).pcid
-        const ppid = getQueryParams(location.href).ppid
-        if (pcid || ppid){
-          let tempHeader = {...header}
-          if(pcid)
-            tempHeader.pcid=pcid
-          if(ppid)
-            tempHeader.ppid=ppid
-          setHeader(() => (tempHeader));
+    const geneId = location.hash.split('/')[2];
```

Comment:
Add a comment explaining that the `location.hash` is expected to be in the form of `<catalog-id>/<schema>:<table>/NCBI_GeneID=<value>`. Also that this works specifically for genes and their associated data in the  `Array_Data_view`.